### PR TITLE
Refactored the QueryTypeContentViewMapper

### PR DESCRIPTION
**Pull request by bdunogier**
`QueryTypeContentViewMapper` is now an interface, in order to document both the input and output expectations.

The existing implementation is renamed to `QueryParameterContentViewQueryTypeMapper`, as it maps using the `query` key from the view configuration parameters.

### CODE REVIEW

File: eZ/Publish/Core/QueryType/ContentViewQueryTypeMapper.php [hunk 1]
```diff
@@ -7,78 +7,7 @@
  */
 namespace eZ\Publish\Core\QueryType;
 
-use eZ\Publish\Core\MVC\Symfony\View\ContentView;
-use InvalidArgumentException;
-use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
-
-class ContentViewQueryTypeMapper implements QueryTypeMapper
+interface QueryTypeMapper
```
Comment (andrerom) [inline eZ/Publish/Core/QueryType/ContentViewQueryTypeMapper.php]: ContentViewQueryTypeMapper

File: eZ/Publish/Core/QueryType/ContentViewQueryTypeMapper.php [hunk 2]
```diff
@@ -7,78 +7,7 @@
...
-                    'view' => $contentView,
-                    'location' => $contentView->getLocation(),
-                    'content' => $contentView->getContent(),
-                ]
-            );
-        } else {
-            return $queryParameterValue;
-        }
-    }
+    public function map($input);
```
Comment (andrerom) [inline eZ/Publish/Core/QueryType/ContentViewQueryTypeMapper.php]: type hints and phpdoc _(param, return type, desc, bonus: example use)_?

File: eZ/Publish/Core/QueryType/ContentViewQueryTypeMapper.php (hunk 1 above)
Comment (bdunogier) [inline eZ/Publish/Core/QueryType/ContentViewQueryTypeMapper.php]: Yeah, the change didn't make it in the commit. Hence the failure :)

**Comment by andrerom** [comment]
+1

**Pull request merged**